app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -24,23 +40,99 @@ open class FeedViewModel : ViewModel() {
     }
 
     fun refresh() {
-        //todo: fix the implementation
-        isLoading.value = false
-        isEmpty.value = true
+        loadData()
     }
+
+
+    private var compositeDisposable: CompositeDisposable? = null
+
+    private fun loadData() {
+        val moshi = Moshi.Builder()
```

Comment:
`Moshi` and `Retrofit` don't need to be recreated every time user pulls to refresh.
You can make them member variables of `FeedViewModel` and initialise them once.